### PR TITLE
🎉 Source Amazon Ads: add option to customize look back window value

### LINKED ISSUE #18929: Amazon Ads Source - make lookback window customizable
## Tell us about the problem you're trying to solve
I want to be able to always resync the past 30 days of data on each sync with this source without having to change the start date each time

## Describe the solution you’d like
Make the lookback window customizable for Amazon Ads source 

## Describe the alternative you’ve considered or used
Changing the start date each sync as the day changes 

## Additional context
-
## Are you willing to submit a PR?
-

### PR DESCRIPTION
**Pull request by dandpz**
resolve #18929 
## What
1. Add "look_back_window" internal settings
2. Add "Look Back Window" to source settings in order to let the user customize this value

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
The User will find a new field to set inside the resource settings. No breaking change, default value is the old one.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

**Comment by marcosmarxm** [comment]
Hello :wave::skin-tone-2: and thank you for your contribution!

Airbyte has instituted a code freeze between 19 and 30 December, to make sure there are no disruptions during the holidays. 
Because of this, reviewing and merging your contribution may take longer than usual. 
We apologize for the delay, but we want everyone to have a quiet and happy holiday.

If you have any questions or need further clarification, please don't hesitate to ping via Slack.

**Review by marcosmarxm — APPROVED** [review]
Thanks @dandpz I only need the final approval from the connector team.

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py
Comment (grubberr) [inline airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py:332]: can you remove this constant from 
```
LOOK_BACK_WINDOW = 3
```

it seems we don't need it now

**Comment by grubberr** [comment]
@dandpz can you please merge master branch and resolve conflicts?

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml
Comment (grubberr) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:60]: you don't need to bump version here, it's done automatically by publishing scripts

File: airbyte-config/init/src/main/resources/seed/source_specs.yaml
Comment (grubberr) [inline airbyte-config/init/src/main/resources/seed/source_specs.yaml:703]: you don't need to bump version here, it's done automatically by publishing scripts

File: airbyte-config/init/src/main/resources/seed/source_specs.yaml
Comment (grubberr) [inline airbyte-config/init/src/main/resources/seed/source_specs.yaml:779]: you don't need to touch this file, it's done automatically by publishing scripts

**Review by grubberr — APPROVED** [review]

**Comment by grubberr** [comment]
@dandpz I have approved
but revert changes on `source_specs.yaml` and `source_definitions.yaml`
this files updated automatically

**Comment by dandpz** [comment]
> @dandpz I have approved but revert changes on `source_specs.yaml` and `source_definitions.yaml` this files updated automatically

Ok, I am reverting them, sorry I wasn't aware of the automatic update of the files

**Comment by grubberr** [comment]
@YowanR @lazebnyi new config parameter was introduced to the source-amazon-ads GA connector can you please look?

**Review by grubberr — CHANGES_REQUESTED** [review]
sorry I missed this

can you please pre-load default value 
like did here for "region"

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/source.py#L52

**Comment by grubberr** [comment]
@dandpz please fully revert `source_specs.yaml` not only version.
it's updated automatically

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py
Comment (grubberr) [inline airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py:119]: because `look_back_window` not optional can you please 
change `config['look_back_window']`
?

**Review by grubberr — APPROVED** [review]

**Comment by grubberr** [comment]
@dandpz it seems unit-tests failed

**Review by grubberr — CHANGES_REQUESTED** [review]
it seems unit-tests failed

**Review by grubberr — APPROVED** [review]

**Pull request merged**
